drivers/openstack/provider.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018 Drone.IO Inc
+// Use of this software is governed by the Business Source License
+// that can be found in the LICENSE file.
+
+package openstack
+
+import (
+	"sync"
+	"text/template"
+
+	"github.com/drone/autoscaler"
+	"github.com/drone/autoscaler/drivers/internal/userdata"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+)
+
+// provider implements an OpenStack provider
+type provider struct {
+	init sync.Once
+
+	key      string
+	region   string
+	image    string
+	flavor   string
+	subnet   string
+	pool     string
+	userdata *template.Template
+	groups   []string
+	metadata map[string]string
+
+	computeClient *gophercloud.ServiceClient
+}
+
+// New returns a new OpenStack provider.
+func New(opts ...Option) autoscaler.Provider {
+	p := new(provider)
+	for _, opt := range opts {
+		opt(p)
+	}
+
+	if p.userdata == nil {
+		p.userdata = userdata.T
+	}
+
+	if p.computeClient == nil {
+		opts, err := openstack.AuthOptionsFromEnv()
```

Comment:
You should use a different name here, opts is already an argument of this function.